src/Toml/Parser.hs

```diff
@@ -122,55 +119,23 @@ keyValP = do
 
 data TableHeader = TableHeader
     { thName :: Key
-    , thKeys :: [(Key, AnyValue)]
+    , thKeys :: TOML
     }
 
 tableHeaderP :: Parser TableHeader
-tableHeaderP = liftA2 TableHeader tableNameP (many keyValP)
+tableHeaderP = liftA2 TableHeader tableNameP (makeToml <$> many keyValP)
+  where
+    makeToml :: [(Key, AnyValue)] -> TOML
+    makeToml kv = TOML (HashMap.fromList kv) mempty
 
 tomlP :: Parser TOML
 tomlP = do
     kvs    <- many keyValP
     tables <- many tableHeaderP
     pure TOML { tomlPairs  = HashMap.fromList kvs
-              , tomlTables = merge $ List.sortBy (comparing thName) tables
+              , tomlTables = foldr (\TableHeader{..} -> insert thName thKeys) mempty tables
```

Comment:
It's usually better to use `foldl'` function. I think you can use `foldl'` and implement this lambda in `where`. Or maybe you can somehow use `fromList` function here by remove `TableHeader` data type and replacing it with just a pair...

Reply:
That actually cool idea! Will work on it :wink: